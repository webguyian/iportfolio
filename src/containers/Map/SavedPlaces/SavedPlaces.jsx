import React from 'react';
import PropTypes from 'prop-types';

import Button from 'components/Button/Button';
import Link from 'components/Link/Link';
import Text from 'components/Text/Text';

const SavedPlaces = props => {
  const { onClick, places } = props;
  const baseClass = 'saved-places';
  const listClass = `${baseClass}-list`;

  return (
    <div className={baseClass}>
      <Text element="h1" type="display">
        Saved
      </Text>
      {places.length ? (
        <ul className={listClass}>
          {places.map(place => (
            <li key={place.id} className={`${listClass}-item`}>
              <Button
                className={`${listClass}-item-btn`}
                onClick={onClick.bind(null, place)}
              >
                <Text element="h2" modifier="bold">
                  {place.name}
                </Text>
                <Text modifier="block">{place.formatted_address}</Text>
                {place.website && (
                  <Link
                    to={place.website}
                    onClick={e => e.stopPropagation()}
                    external
                  >
                    View website
                  </Link>
                )}
              </Button>
            </li>
          ))}
        </ul>
      ) : (
        <div className={`${baseClass}-empty-message`}>
          <Text element="p">You have no saved places.</Text>
          <Text element="p">
            Explore the map by searching and star your favorites.
          </Text>
        </div>
      )}
    </div>
  );
};

SavedPlaces.propTypes = {
  onClick: PropTypes.func.isRequired,
  places: PropTypes.array.isRequired
};

export default SavedPlaces;
