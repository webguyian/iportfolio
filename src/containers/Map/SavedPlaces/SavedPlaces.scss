/* stylelint-disable color-no-hex */
$map-bg-color: #f2f3f4;

.saved-places {
  @include stretch-column-nowrap;
  z-index: 2;
  position: absolute;
  top: 0;
  height: $map-height;
  padding: 40px 0 $gutter;
  background-color: $map-bg-color;
  color: $gray-dark;

  > .ui-text--display {
    margin: $gutter;
  }

  &-list {
    @include unstyled-list;

    &-item {
      &:nth-child(odd) {
        background: $gray-light;
      }

      &-btn {
        display: block;
        width: 100%;
        padding: $gutter;
        text-align: left;

        > .ui-text {
          margin: 0 0 $gutter-xs;
        }

        > .ui-link {
          @include type-body;
          color: $blue;
        }
      }
    }
  }

  &-empty-message {
    min-height: 40%;
    margin: auto $gutter;
    text-align: center;

    > .ui-text--body {
      font-size: $font-size-l;
      line-height: 1.5;
    }
  }
}
