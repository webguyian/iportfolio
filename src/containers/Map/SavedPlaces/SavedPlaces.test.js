/* eslint-disable camelcase */
import React from 'react';
import { act, create } from 'react-test-renderer';

import SavedPlaces from './SavedPlaces';

describe('<SavedPlaces />', () => {
  const props = {
    onClick: jest.fn(),
    places: [
      {
        id: 'a',
        name: 'Place A',
        formatted_address: '123 Apple Road'
      },
      {
        id: 'b',
        name: 'Place B',
        formatted_address: '456 Cherry Lane',
        website: 'http://theplace.to.b'
      }
    ]
  };

  it('renders correctly', () => {
    const component = create(<SavedPlaces {...props} />);

    expect(component).toMatchSnapshot();
  });

  it('renders correctly without places', () => {
    const component = create(<SavedPlaces {...props} places={[]} />);

    expect(component).toMatchSnapshot();
  });

  it('handles link click', () => {
    const component = create(<SavedPlaces {...props} />);
    const [link] = component.root.findAllByType('mock-link');
    const mockEvent = { stopPropagation: jest.fn() };

    expect(mockEvent.stopPropagation).not.toHaveBeenCalled();

    act(() => {
      link.props.onClick(mockEvent);
    });

    expect(mockEvent.stopPropagation).toHaveBeenCalled();
  });
});
