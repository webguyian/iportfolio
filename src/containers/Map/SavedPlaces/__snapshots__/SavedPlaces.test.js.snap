// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SavedPlaces /> renders correctly 1`] = `
<div
  className="saved-places"
>
  <h1
    className="ui-text ui-text--display"
  >
    Saved
  </h1>
  <ul
    className="saved-places-list"
  >
    <li
      className="saved-places-list-item"
    >
      <button
        className="ui-btn saved-places-list-item-btn"
        onClick={[Function]}
        type="button"
      >
        <h2
          className="ui-text ui-text--body ui-text--bold"
        >
          Place A
        </h2>
        <span
          className="ui-text ui-text--body ui-text--block"
        >
          123 Apple Road
        </span>
      </button>
    </li>
    <li
      className="saved-places-list-item"
    >
      <button
        className="ui-btn saved-places-list-item-btn"
        onClick={[Function]}
        type="button"
      >
        <h2
          className="ui-text ui-text--body ui-text--bold"
        >
          Place B
        </h2>
        <span
          className="ui-text ui-text--body ui-text--block"
        >
          456 Cherry Lane
        </span>
        <mock-link
          external={true}
          onClick={[Function]}
          to="http://theplace.to.b"
        >
          View website
        </mock-link>
      </button>
    </li>
  </ul>
</div>
`;

exports[`<SavedPlaces /> renders correctly without places 1`] = `
<div
  className="saved-places"
>
  <h1
    className="ui-text ui-text--display"
  >
    Saved
  </h1>
  <div
    className="saved-places-empty-message"
  >
    <p
      className="ui-text ui-text--body"
    >
      You have no saved places.
    </p>
    <p
      className="ui-text ui-text--body"
    >
      Explore the map by searching and star your favorites.
    </p>
  </div>
</div>
`;
